docs(configuration): add warning for bail and watch (#4889)
webpack/webpack.js.org: src/content/configuration/other-options.mdx

This will force webpack to exit its bundling process.

> W> You can use `bail` option in [`watch`](/configuration/watch) mode, but it will still force webpack to exit as soon as an error is found.

Reviewer: ```suggestion
W> Avoid using `bail` option in [`watch`](/configuration/watch) mode, as it will force webpack to exit as soon as possible when an error is found.
```
